src/mapbox/extended_geolocate_control.js

```diff
@@ -1,12 +1,24 @@
+const Telemetry = require('../libs/telemetry').default
+
 const { GeolocateControl } = require('mapbox-gl--ENV')
 
 /**
 * Override default GeolocateControl
 */
+
+const geolocationPermissions = {
+  PROMPT : 'prompt',
+  GRANTED : 'granted',
+  DENIED : 'denied'
+}
+
 class ExtendedGeolocateControl extends GeolocateControl {
   constructor(options, container) {
     super(options)
     this._container = container
+    this.on('trackuserlocationstart', () => {
+      Telemetry.add(Telemetry.LOCALISE_TRIGGER)
```

Comment:
I am not sure this is a very relevant metric. If location is refused or blocked for some reason, we'd still count each tap on the button.  
Why not triggering this event on the first click only ?

Reply:
i propose a new telemetry method to complete add() : addOnce().